### ISSUE TITLE
Duo Universal Prompt support

### ISSUE THREAD

**Issue by clstrickland** [goauthentik/authentik#5366]
**Is your feature request related to a problem? Please describe.**
Not a problem, but an improvement. Duo Universal Prompt is another way to approach MFA, where MFA is handled entirely by Duo then Authentik would be notified that authentication succeeded. The main benefit of this is that you could "remember" MFA for all services.

Let's say you implement Duo in multiple services, Authentik and Guacamole. So that you can leverage credential passing, you use LDAP in Guacamole, with Duo 2FA. In this situation, Guacamole does LDAP itself, then handles Duo itself using Traditional Prompt (although hopefully they upgrade to Universal Prompt soon). You have Duo remember you for 24 hours.
If Authentik had Universal Prompt, and you had already logged in once that day, signing in to Guacamole would realize you had already done MFA and bypass it.
But, since Authentik doesn't have Duo Universal Prompt, you have to do MFA again.

TLDR:
WIth Universal Prompt, MFA would be stored in the browser session, and one MFA transaction (if desired) could would for all services, instead of Authentik needing to do MFA separately.

**Describe the solution you'd like**
I'm not going to go into details about what Duo Universal Prompt is, because Duo has [an amazing article about it](https://guide.duo.com/universal-prompt), and a more technical explanation [here](https://duo.com/docs/duoweb#overview).
It is built on OIDC, and Duo makes a Python module for this. There is of course a lot that happens in the background but these are the important parts:

Sending the user to Duo Universal Prompt:

    state = duo_client.generate_state()
    session['state'] = state
    session['username'] = username
    prompt_uri = duo_client.create_auth_url(username, state)
    return redirect(prompt_uri)

This would send the user to Duo Universal Prompt. Then, there would have to be an endpoint in Authentik for the Duo callback. It would handle data like so:

    state = request.args.get('state')
    code = request.args.get('duo_code')
    username = session['username']
    # You would check the state to make sure it matches the state you sent
    decoded_token = duo_client.exchange_authorization_code_for_2fa_result(code, username)
    # Step 3: Profit

The `decoded_token` line would throw an error if something was wrong with MFA, so if we make it past that line, MFA worked and we can let the user in. Of course lots of edge cases and error catching needs to be added.

From what I can tell, the hard part here will be handling the callback, since AFAIK Authentik doesn't have anything really like this yet, the closest thing is federated logins.

**Describe alternatives you've considered**
The alternative is of course using the current Duo implementation, but that's why I'm here. I use Universal Prompt at work, and I personally hate using the handful of services we have which don't support this.

**Comment by BeryJu** [comment]
The main reason why authentik doesn't use the universal or traditional prompts is because both of them require a browser environment to run. The current implementation uses the Duo API and as such is usable in environments without a Browser, i.e. authentik's LDAP and Radius servers.

You can also configure authentik to remember MFA by setting `Last validation threshold` in the authenticator validation stage, and have guacamole use authentik to authenticate.

**Comment by jamie-dit** [comment]
+1 for an option to allow Universal Prompt rather than Auth API.

Duo Paid plans allows the installation of a Duo Desktop agent to help verify requests which will improve the workflow for users with Duo significantly
